Mathlib/GroupTheory/Perm/DomMulAct.lean

```diff
@@ -0,0 +1,114 @@
+
+* `DomMulAct.stabilizerMulEquiv` is the `MulEquiv` from
+  the MulOpposite of this stabilizer to the product,
+  for `i : ι`, of `Equiv.Perm {a // p a = i}`.
+
+* Under `Fintype α` and `Fintype ι`, `DomMulAct.stabilizer_card p` computes
+  the cardinality of the type of permutations preserving `p` :
+  `Fintype.card {f : Perm α // p ∘ f = p} =
+    ∏ i, (Fintype.card {a // p a = i}).factorial `
+
+-/
+
+variable {α ι : Type*} {p : α → ι}
+
+open Equiv MulAction
+
+namespace DomMulAct
+
+lemma mem_stabilizer_iff {g : (Perm α)ᵈᵐᵃ} :
+    g ∈ stabilizer (Perm α)ᵈᵐᵃ p ↔ p ∘ (mk.symm g :) = p := by
+  simp only [MulAction.mem_stabilizer_iff]; rfl
+
+/-- The `invFun` component of `MulEquiv` from `MulAction.stabilizer (Perm α) p`
+  to the product of the `Equiv.Perm {a // p a = i} -/
+def stabilizerEquiv_invFun (g : ∀ i, Perm {a // p a = i}) (a : α) : α := g (p a) ⟨a, rfl⟩
+
+lemma stabilizerEquiv_invFun_eq (g : ∀ i, Perm {a // p a = i}) {a : α} {i : ι} (h : p a = i) :
+    stabilizerEquiv_invFun g a = g i ⟨a, h⟩ := by subst h; rfl
+
+lemma comp_stabilizerEquiv_invFun (g : ∀ i, Perm {a // p a = i}) (a : α) :
+    p (stabilizerEquiv_invFun g a) = p a :=
+  (g (p a) ⟨a, rfl⟩).prop
+
+/-- The `invFun` component of `MulEquiv` from `MulAction.stabilizer (Perm α) p`
+  to the product of the `Equiv.Perm {a | p a = i} (as an `Equiv.Perm α`) -/
+def stabilizerEquiv_invFun_aux (g : ∀ i, Perm {a // p a = i}) : Perm α where
+  toFun := stabilizerEquiv_invFun g
+  invFun := stabilizerEquiv_invFun (fun i ↦ (g i).symm)
+  left_inv a := by
+    rw [stabilizerEquiv_invFun_eq _ (comp_stabilizerEquiv_invFun g a)]
+    exact congr_arg Subtype.val ((g <| p a).left_inv _)
+  right_inv a := by
+    rw [stabilizerEquiv_invFun_eq _ (comp_stabilizerEquiv_invFun _ a)]
+    exact congr_arg Subtype.val ((g <| p a).right_inv _)
+
+variable (p)
+
+/-- The `MulEquiv` from the `MulOpposite` of `MulAction.stabilizer (Perm α)ᵈᵐᵃ p`
+  to the product of the `Equiv.Perm {a // p a = i}` -/
+def stabilizerMulEquiv : (stabilizer (Perm α)ᵈᵐᵃ p)ᵐᵒᵖ ≃* (∀ i, Perm {a // p a = i}) where
+  toFun g i := Perm.subtypePerm (mk.symm g.unop) fun a ↦ by
+    rw [← Function.comp_apply (f := p), mem_stabilizer_iff.mp g.unop.prop]
+  invFun g := ⟨mk (stabilizerEquiv_invFun_aux g), by
+    ext a
+    rw [smul_apply, symm_apply_apply, Perm.smul_def]
+    apply comp_stabilizerEquiv_invFun⟩
+  left_inv g := rfl
+  right_inv g := by ext i a; apply stabilizerEquiv_invFun_eq
+  map_mul' g h := by rfl
+
+variable {p}
+
+lemma stabilizerMulEquiv_apply (g : (stabilizer (Perm α)ᵈᵐᵃ p)ᵐᵒᵖ) {a : α} {i : ι} (h : p a = i) :
+    ((stabilizerMulEquiv p)) g i ⟨a, h⟩ = (mk.symm g.unop : Equiv.Perm α) a := rfl
+
+section Fintype
+
+variable [Fintype α] [Fintype ι] [DecidableEq α] [DecidableEq ι]
+
+open BigOperators
+
+variable (p)
+
+/-- The cardinality of the type of permutations preserving a function -/
+theorem stabilizer_card:
+    Fintype.card {f : Perm α // p ∘ f = p} =
```

Comment:
It's a bit confusing here that `p` is the **f**unction and `f` is the **p**ermutation,

Reply:
In my mind, p was the projection of a fibration and the permutations are called f all over mathlib.
I'll make the change.

Reply:
It might be worth waiting for a second opinion; an alterative would be to change the docstring to use the word `fibration` somehow

Reply:
I have done it already, that was quick…